pkg/cmd/agent/restore.go

```diff
@@ -0,0 +1,173 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-chi/render"
+	"github.com/pkg/errors"
+	"github.com/scylladb/go-log"
+	"github.com/scylladb/scylla-manager/v3/pkg/config/agent"
+	"github.com/scylladb/scylla-manager/v3/swagger/gen/agent/models"
+)
+
+type restoreHandler struct {
+	config agent.Config
+	logger *log.Logger
+}
+
+func newRestoreHandler(c agent.Config) *restoreHandler {
+	return &restoreHandler{config: c}
+}
+
+// restore SSTables that are already downloaded to table upload directory.
+func (h *restoreHandler) restore(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	// Initialise logger
+	if h.logger == nil {
+		l, err := h.config.MakeLogger()
+		if err != nil {
+			h.respondWithInternalError(ctx, w, r, errors.Wrap(err, "create logger"))
+
+			return
+		}
+
+		h.logger = &l
+	}
+
+	h.logger.Info(ctx, "Restoring files")
+
+	params := new(models.RestoreParams)
+	dec := json.NewDecoder(r.Body)
+	err := dec.Decode(params)
+	if err != nil {
+		h.respondWithInternalError(ctx, w, r, errors.Wrap(err, "decode params"))
+
+		return
+	}
+
+	h.logger.Info(ctx, "Restore params",
+		"keyspace", params.Keyspace,
+		"table", params.Table,
+		"version", params.Version,
+		"files", params.Files,
+	)
+
+	uploadDir := uploadDataDir(h.config.Scylla.DataDirectory, params.Keyspace, params.Table, params.Version)
+	absUploadDir, err := filepath.Abs(uploadDir)
+	if err != nil {
+		h.respondWithInternalError(ctx, w, r, errors.Wrap(err, "get upload directory absolute path"))
+
+		return
+	}
+
+	h.logger.Info(ctx, "Upload data directory",
+		"dir", absUploadDir,
+	)
+
+	//TODO - how to do this ownership in production? Should the manager be running as scylla user?
+	for _, f := range params.Files {
+		h.logger.Info(ctx, "Restore file",
+			"file", f,
+		)
+
+		err = os.Chown(
+			path.Join(absUploadDir, f),
+			107,
+			109,
+		)
```

Comment:
What about using using chmod and setting read (or write) permissions to others with https://pkg.go.dev/os#Chmod ?
I think it's much easier than trying to find the uid and gid of scylla user in system and changing the owner.
The only concern about chmod with +r is the security. I'm not an expert there, but I think if it's in the scylla owned directory then no other user will see the file anyway.

Reply:
Unfortunately even with `os.Chomd("file", 0777)` the same error is returned - Scylla checks for file ownership (probably because it wants to delete the file after issuing load and stream).

Reply:
Since the ownership problem occurs only with Scylla 4.6.1 (or older) that does not have to be supported by the restore, we shouldn't change ownership of the downloaded files. Having said that, we should bump current Scylla version used for testing.
I tried to bump Scylla version to 5.x, but currently it's not possible because of #3235, so for now I will bump it to 4.6.7.